mlir/lib/Dialect/Vector/IR/VectorOps.cpp

```diff
@@ -2385,9 +2386,105 @@ static LogicalResult rewriteFromElementsAsSplat(FromElementsOp fromElementsOp,
   return success();
 }
 
+/// Rewrite vector.from_elements as vector.shape_cast, if possible.
```

Comment:
[ultra nit] "if possible" is implicit and "less is more" :)
```suggestion
/// Rewrite vector.from_elements as vector.shape_cast.
```